### ISSUE TITLE
Open TDirectoryFile

### ISSUE THREAD

**Issue by l-althueser** [scikit-hep/uproot#64]
I was trying to open a regular ROOT file with the following structure:
```
KEY: TDirectoryFile   events;1        events

TDirectoryFile*         events  events
 KEY: TTree     events;1       Tree containing event data
```

If I try to open the file and change the directory with
```
f = uproot.open('test.root')['events']
```
I only get a `<Undefined (no class named 'TDirectoryFile') at 0x7f5cc9f9d0b8>`.

I have also tried to rename the TTree to something else that is not identical to the TDirectory name and this works as expected! So uproot seems to have a problem if the TDirectory and TTree in it have the same name.

**Comment by jpivarski** [comment]
I didn't know that two objects could have the same name at the same level! Could you send me an example file so that I can look at it in detail? (If it's small enough and can be made public, I'll add it as a unit test so that this bug doesn't regress.)

**Comment by l-althueser** [comment]
Sorry, I thought that the problem would be the naming. But I generated now two small test files (explained in the following) and both seem to work - I am a bit puzzled what happened. The issue could also be that I try to load a ROOT 5 file and test with ROOT 6. Are there known incompatibilities? I could send you the file I want to load in private (500MB) .. in the meantime I will continue to figure out where the issue exactly is.

This is what I tested and what I can confirm that both worked:

File 1:
"events" - TDirectory
 |-> "events" TTree (within the TDirectory)

File 2:
"events" - TDirectory
 |-> "Tevents" TTree (within the TDirectory)

**Comment by jpivarski** [comment]
In principle, ROOT 5 and 6 should be compatible. You're talking about writing files in ROOT 5 or 6 and reading in uproot, right?

(I've got a series of unit tests written in all ROOT versions from the past 5 years, which includes several versions of ROOT 5. But there can always be corner cases that I didn't explicitly test...)

It should be perfectly legal to name an object inside a directory the same name as the directory. Your second tests behave as expected. What I was saying is that I didn't think it was legal to have two objects at the same level with the same name. (After all, how would you extract them by name?)

You can reopen this issue if you reproduce the error.

**Comment by jpivarski** [comment]
Okay, I'll take a look.

**Comment by l-althueser** [comment]
Thanks! Just for reference, I tried to load with:
```
import uproot
f = uproot.open('test_same_names.root')['events']
f.keys()
```

**Comment by jpivarski** [comment]
In uproot, `TFile` and `TDirectory` were the only exceptions to Python classes exactly mirroring C++ classes. (Technical reason: I couldn't see a way to do it without repeating a lot of code.) `TDirectoryFile` should have been another exception. All I had to do was replace the

```python
if classname == "TDirectory":
```

special cases with

```python
if classname == "TDirectory" or classname == "TDirectoryFile":
```

They have the same serialization and the same meaning. This fix is going in as version 2.8.11 (in master now; in the process of deployment to PyPI).

**Comment by l-althueser** [comment]
Many thanks for the quick fix!
